packages/perseus/src/widgets/expression/expression.util.ts

```diff
@@ -0,0 +1,36 @@
+import type {
+    LegacyButtonSets,
+    PerseusExpressionWidgetOptions,
+} from "@khanacademy/perseus-core";
+
+/**
+ * For details on the individual options, see the
+ * PerseusExpressionWidgetOptions type
+ */
+type ExpressionPublicWidgetOptions = {
+    buttonSets: LegacyButtonSets;
+    functions: ReadonlyArray<string>;
+    times: boolean;
+    visibleLabel?: string;
+    ariaLabel?: string;
+    buttonsVisible?: "always" | "never" | "focused";
+};
```

Comment:
PublicWidgetOptions will always be a subset of WidgetOptions, so I think it would make sense to derive one from the other.

```suggestion
type ExpressionPublicWidgetOptions = {
    buttonSets: ExpressionWidgetOptions["buttonSets"];
    functions: ExpressionWidgetOptions["functions"];
    times: ExpressionWidgetOptions["times"];
    visibleLabel?: ExpressionWidgetOptions["visibleLabel"];
    ariaLabel?: ExpressionWidgetOptions["ariaLabel"];
    buttonsVisible?: ExpressionWidgetOptions["buttonsVisible"];
};
```

This will get more complicated for nested data. For instance if we're trying to remove `correct` from `options.choices.correct`, we can't do this exactly; but for a lot of this stuff I think we'd be fine.

If someone else (or past me) pitched a reason this wouldn't work, feel free to ignore.

Reply:
Updated :)